examples/test.cpp

```diff
@@ -1,6 +1,33 @@
-#include "../include/featurelib.h"
-
+#include "../include/FeatureLib.h"
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/highgui.hpp>
+std::vector<uint8_t> matToVector(const cv::Mat &mat) {
+  std::vector<uint8_t> result;
+  cv::Mat cont = mat.isContinuous() ? mat : mat.clone();  // make sure the Mat is continuous
+  cont.reshape(1, 1).copyTo(result);  // copy from a 1D Mat
+  return result;
+}
 int main() {
-  featurelib instance;
+  std::string image_path = cv::samples::findFile("church.jpg");
```

Comment:
It would be better to accept an input from command line and pick a file chosen by a user

Reply:
done, but now my test fails:
cv::cvtColor(Image, Image, cv::COLOR_BGR2GRAY);  with that line test isn't working
> Invalid number of channels in input image:
>     'VScn::contains(scn)'
> where
>     'scn' is 1

Reply:
You can debug this by checking Mat type of `Image`. You'll probably need this: https://gist.github.com/yangcha/38f2fa630e223a8546f9b48ebbb3e61a

Reply:
Yep, i fix it